packages/react-ui-core/src/DatePicker/DatePicker.js

```diff
@@ -79,72 +58,92 @@ export default class DatePicker extends Component {
     document.removeEventListener('click', this.onOutsideClick, false)
   }
 
+  @autobind
   onOutsideClick(event) {
-    if (!this.mainNode.contains(event.target) && this.state.showCalendar) {
+    if (!this.picker.current.contains(event.target) && this.state.showCalendar) {
```

Comment:
Will `picker.current` always exist?

Reply:
Yes... it's a reference.  That's the way you reference a react component that's been mounted.